File: 2487-optimal-partition-of-string/optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        seen = set()
        count = 0 
        for c in s:
            if c in seen:
                count += 1
                seen = set()
            seen.add(c)
        if seen:
            count += 1
        return count

File: 2487-optimal-partition-of-string/optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        int i = 0;
        int n = s.length();
        map<char, int>m;
        int count = 0;
        while(i<n)
        {
            if(m[s[i]] == 0)
            {
                m[s[i]] = 1;
                i++;
            }
            else
            {
                count++;
                m.clear();
            }
        }
        return count + 1;
    }
};